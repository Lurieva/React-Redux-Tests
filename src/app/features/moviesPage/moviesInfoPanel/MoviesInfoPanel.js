import React from 'react';
import PropTypes from 'prop-types';

import './MoviesInfoPanel.scss';

const MoviesInfoPanel = ({ count, children }) => (
    <div className="moviesInfoPanel">
        <div className="moviesInfoPanel__count">
            { count === 0 && <span></span> }
            { count === 1 && <span>{count} movie found</span> }
            { count > 1 && <span>{count} movies found</span> }
        </div>
        <div className="moviewInfoPanel__sortBy">
            { children }
        </div>
    </div>
);

MoviesInfoPanel.propTypes = {
    count: PropTypes.number,
    children: PropTypes.element
};

export default MoviesInfoPanel;