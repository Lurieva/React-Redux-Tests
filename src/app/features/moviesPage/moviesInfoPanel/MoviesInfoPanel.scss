.moviesInfoPanel {
    display: flex;
    justify-content: space-between;

    &__count {
        font-weight: bold;
    }
}