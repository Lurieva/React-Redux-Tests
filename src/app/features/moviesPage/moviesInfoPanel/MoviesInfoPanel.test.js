import React from 'react';
import { shallow, containsMatchingElement } from 'enzyme';
import MoviesInfoPanel from './MoviesInfoPanel';

describe('MoviesInfoPanel', () => {
    it('should be defined', () => {
        expect(MoviesInfoPanel).toBeDefined();
    });

    it('should render correctly ', () => {
        const tree = shallow(<MoviesInfoPanel />);
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly if count === 0', () => {
        const tree = shallow(<MoviesInfoPanel count={0}/>);
        expect(
            tree.containsMatchingElement(
              <span></span>
            )
        ).toBeTruthy()
    });

    it('should render correctly if count === 1', () => {
        const tree = shallow(<MoviesInfoPanel count={1}/>);
        expect(
            tree.containsMatchingElement(
                <span>1 movie found</span>
            )
        ).toBeTruthy()
    });

    it('should render correctly if count > 1', () => {
        const tree = shallow(<MoviesInfoPanel count={10}/>);
        expect(
            tree.containsMatchingElement(
                <span>10 movies found</span>
            )
        ).toBeTruthy()
    });
});