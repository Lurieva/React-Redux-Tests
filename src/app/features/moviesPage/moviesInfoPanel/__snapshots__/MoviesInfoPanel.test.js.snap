// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesInfoPanel should render correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviesInfoPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="moviesInfoPanel__count"
        />,
        <div
          className="moviewInfoPanel__sortBy"
        />,
      ],
      "className": "moviesInfoPanel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            false,
            false,
          ],
          "className": "moviesInfoPanel__count",
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "moviewInfoPanel__sortBy",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="moviesInfoPanel__count"
          />,
          <div
            className="moviewInfoPanel__sortBy"
          />,
        ],
        "className": "moviesInfoPanel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              false,
              false,
            ],
            "className": "moviesInfoPanel__count",
          },
          "ref": null,
          "rendered": Array [
            false,
            false,
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "moviewInfoPanel__sortBy",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
